test/Microsoft.AspNet.Mvc.Core.Test/DefaultControllerTypeProviderTest.cs

```diff
@@ -164,6 +178,55 @@ public void IsController_NoControllerSuffix()
             Assert.True(isController);
         }
 
+        public static TheoryData TypesWithPocoControllerAncestor
+        {
+            get
+            {
+                return new TheoryData<TypeInfo>
+                {
+                   typeof(ChildWithoutSuffix).GetTypeInfo(),
+                   typeof(DescendantLevel1).GetTypeInfo(),
+                   typeof(DescendantLevel2).GetTypeInfo()
+                };
+            }
+        }
+
+        [Theory]
+        [InlineData(typeof(ChildWithoutSuffix))]
+        [InlineData(typeof(DescendantLevel1))]
+        [InlineData(typeof(DescendantLevel2))]
+        public void IsController_ReturnsTrue_IfAncestorTypeNameHasControllerSuffix(Type type)
+        {
+            // Arrange
+            var provider = GetControllerTypeProvider();
+
+            // Act
+            var isController = provider.IsController(type.GetTypeInfo());
+
+            // Assert
+            Assert.True(isController);
+        }
+
+        [Theory]
+        [InlineData(typeof(BaseNonControllerController))]
+        [InlineData(typeof(BaseNonControllerControllerChild))]
+        [InlineData(typeof(BasePocoNonControllerController))]
+        [InlineData(typeof(BasePocoNonControllerControllerChild))]
+        [InlineData(typeof(NonController))]
+        [InlineData(typeof(NonControllerChild))]
+        [InlineData(typeof(PersonModel))]
```

Comment:
It's to verify we don't throw in the object graph walk when you don't have controller type in your hierarchy (it's the only scenario where we'll exit the while loop of `DerivesFromController` at the end.